tuf/keys.py

```diff
@@ -692,27 +690,8 @@ def create_signature(
 
     data:
       Data object used by create_signature() to generate the signature.
-
-    force_treat_as_pydict: (optional; default False)
-      If True, then regardless of tuf.conf.METADATA_FORMAT, we will treat the
-      data provided as if it is a Python dictionary that must be canonicalized
-      and then encoded as utf-8 before it is signed.
-      Otherwise, behavior depends on tuf.conf.METADATA_FORMAT. If that is
-      'json', then the canonicalization occurs. If it is instead 'der', then
-      the canonicalization does not occur.
-
-    force_non_json: (optional, default False)
-      If True, then regardless of tuf.conf.METADATA_FORMAT, we will treat the
-      data provided as if it is not a Python dictionary and cannot be
-      canonicalized before it is signed.
-      Whether or not it is encoded as utf-8 depends on the format and
-      is_binary_data.
-
-    is_binary_data: (optional; default False)
-      If True, then regardless of tuf.conf.METADATA_FORMAT, we skip trying to
-      utf-8 encode the data, because it cannot necessarily be encoded as utf-8
-      and in any case is already encoded.
-
+      This should be bytes(). If it is a string, it should already be encoded
```

Comment:
What happens if the string isn't in bytes?  What error should the caller expect?